docs/essentials/overriding.md

```diff
@@ -0,0 +1,151 @@
+# Overriding
+
+Overriding allows for the easy replacement of parts of SFX, thereby modifying, adding, or removing functionality. For example, you can easily override how adding to cart works, how header/footer looks, how the catalog is displayed etc.
+
+Overriding works in a file-based manner, so basically any file (page, component, service, composable, etc.) can be overridden.
+
+::: tip
+Overriding rules are managed in the `storefront-x.config.js` file based on the position of imported module. The file from a later module will be used instead of the file from an earlier module. If only one module is imported, the file from that module will be used.
+:::
+
+Overriding is one of the primary feature that allows Storefront X to be extremely flexible and performant at the same time.
+
+## How to override?
+
+Imagine we have a default `module-a` with the `Header.vue` component in the `/organisms` concept.
+
+By default, this header contains the default setup, but with `module-b` enabled, this header will be overridden to our setup. Let's imagine our company would like to have a fixed menu all the time while scrolling and a different color palette. The best way to achieve this is to override the component.
+
+```vue
+// module-a/organisms/Header.vue
+
+<template>
+  <header class="bg-white border-b-2 border-primary-400 fixed top-0 md:static z-20 left-0 right-0">
+    <HeaderLinks />
```

Comment:
Again, these our specific components for our demo and 3rd party readers wont know what hey represent. Instead use components like <Logo />, <SearchBart />, <MicroCart /> etc